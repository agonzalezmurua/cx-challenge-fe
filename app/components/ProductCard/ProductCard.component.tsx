import { Product } from "@/models/Product.model";
import { useTranslation } from "next-i18next";
import Image from "next/image";
import { BsTruck } from "react-icons/bs";
import styles from "./ProductCard.module.scss";

type ProductCardProps = Product;

export const ProductCard = ({
  address,
  free_shipping,
  id,
  installments,
  picture,
  price,
  title,
  condition,
}: ProductCardProps) => {
  const { t } = useTranslation("home");

  return (
    <section className={styles.container}>
      <section className={styles.product_picture__container}>
        <Image src={picture} alt="product picture" height={200} width={200} />
      </section>
      <section className={styles.product_container}>
        <section className={styles.product_price__container}>
          <span className={styles.product_price__amount}>
            {t("product.price", {
              amount: price.amount,
              formatParams: {
                amount: { currency: price.currency },
              },
            })}
          </span>
          {free_shipping && (
            <span className={styles.product_price__free_shipping}>
              <BsTruck />
            </span>
          )}
        </section>
        <h2 className={styles.product_name}>{title}</h2>
        <span className={styles.product_installments}>
          {t("product.installments", {
            quantity: installments.quantity,
            amount: installments.amount,
            formatParams: {
              amount: { currency: price.currency },
            },
          })}
        </span>
      </section>
      <section className={styles.product_aside__container}>
        {address.city_name}
      </section>
    </section>
  );
};
