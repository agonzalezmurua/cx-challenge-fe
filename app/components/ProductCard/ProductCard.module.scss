.container {
  display: flex;
  height: 15rem;

  gap: 1rem;

  padding: 1rem 0;
}


.product_container {
  flex: 4;
  
  display: inline-flex;
  flex-direction: column;

  gap: 1rem;
  justify-content: center;
  
}

.product_picture__container {
  position: relative;
  aspect-ratio: 1/1;

  border-radius: var(--border-radius-sm);

  overflow: hidden;
}

.product_price__container {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  font-size: xx-large;
}

.product_price__amount {
  // Keep this in case we want to customize the price tag
}

.product_price__free_shipping {
  font-size: large;
  background-color: var(--green-500);
  color: var(--gray-700);
  display: inline-flex;
  border-radius: 50%;
  padding: 0.25rem;
}

.product_name {
  font-size: x-large;
  font-weight: normal;
}

.product_installments {
  font-size:medium;
  color: var(--green-700);
}

.product_aside__container {
  flex: 1;

  padding-top: 4rem;
  color: var(--gray-500);
  padding-left: 0;
}