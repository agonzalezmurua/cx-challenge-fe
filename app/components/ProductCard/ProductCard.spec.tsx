import { Product } from "@/models/Product.model";
import { render } from "@testing-library/react";
import { ProductCard } from "./ProductCard.component";

const product: Product = {
  address: {
    city_name: "Caballito",
    state_name: "Capital Federal",
  },
  condition: "new",
  free_shipping: true,
  title: "Apple AirPods Con Estuche De Carga - Blanco",
  id: "MLA1363069275",
  installments: {
    amount: "6674.27",
    quantity: 6,
  },
  picture: "http://http2.mlstatic.com/D_923638-MLA54361048207_032023-I.jpg",
  price: {
    currency: "ARS",
    amount: "26994",
    decimals: 0,
  },
};

it("should be defined", () => {
  expect(ProductCard).toBeDefined();
});

it("should render and match snapshot", () => {
  const result = render(<ProductCard {...product} />);

  expect(result.asFragment()).toMatchSnapshot();
  expect(result.getByText(product.title)).toBeInTheDocument();
});
