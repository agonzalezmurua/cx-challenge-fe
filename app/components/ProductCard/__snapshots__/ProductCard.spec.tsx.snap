// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <section
      class="product_picture__container"
    >
      <img
        alt="product picture"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_923638-MLA54361048207_032023-I.jpg&w=640&q=75"
        srcset="/_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_923638-MLA54361048207_032023-I.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_923638-MLA54361048207_032023-I.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </section>
    <section
      class="product_container"
    >
      <section
        class="product_price__container"
      >
        <span
          class="product_price__amount"
        >
          product.price
        </span>
        <span
          class="product_price__free_shipping"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 3.5A1.5 1.5 0 0 1 1.5 2h9A1.5 1.5 0 0 1 12 3.5V5h1.02a1.5 1.5 0 0 1 1.17.563l1.481 1.85a1.5 1.5 0 0 1 .329.938V10.5a1.5 1.5 0 0 1-1.5 1.5H14a2 2 0 1 1-4 0H5a2 2 0 1 1-3.998-.085A1.5 1.5 0 0 1 0 10.5v-7zm1.294 7.456A1.999 1.999 0 0 1 4.732 11h5.536a2.01 2.01 0 0 1 .732-.732V3.5a.5.5 0 0 0-.5-.5h-9a.5.5 0 0 0-.5.5v7a.5.5 0 0 0 .294.456zM12 10a2 2 0 0 1 1.732 1h.768a.5.5 0 0 0 .5-.5V8.35a.5.5 0 0 0-.11-.312l-1.48-1.85A.5.5 0 0 0 13.02 6H12v4zm-9 1a1 1 0 1 0 0 2 1 1 0 0 0 0-2zm9 0a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"
            />
          </svg>
        </span>
      </section>
      <h2
        class="product_name"
      >
        Apple AirPods Con Estuche De Carga - Blanco
      </h2>
      <span
        class="product_installments"
      >
        product.installments
      </span>
    </section>
    <section
      class="product_aside__container"
    >
      Caballito
    </section>
  </section>
</DocumentFragment>
`;
